lib/smack/MemorySafetyChecker.cpp

```diff
@@ -0,0 +1,49 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+#include "smack/MemorySafetyChecker.h"
+#include "smack/Naming.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/Support/Debug.h"
+
+namespace smack {
+
+using namespace llvm;
+
+bool MemorySafetyChecker::runOnModule(Module& m) {
+  DataLayout* dataLayout = new DataLayout(&m);
+  Function* memorySafetyFunction = m.getFunction(Naming::MEMORY_SAFETY_FUNCTION);
+  assert(memorySafetyFunction != NULL && "Couldn't find memory safety function");
+  for (auto& F : m) {
+    if (!Naming::isSmackName(F.getName())) {
+      for (inst_iterator I = inst_begin(F), E = inst_end(F); I != E; ++I) {
+        if (LoadInst* li = dyn_cast<LoadInst>(&*I)) {
+          Value* pointer = li->getPointerOperand();
+          Value* size = ConstantInt::get(IntegerType::get(F.getContext(), 64), 
```

Comment:
No, I don't think you can. The reason is quite simple - SMACK and the target application we want to verify might not be compiled on the same platform. For example, SMACK would likely be compiled on a 64-bit platform, while the target application might be compiled with `-m32` since it is supposed to be running on a 32-bit platform.
Hence, you have to get this size from the target application bitcode file, for example by looking at the size of the second argument of the memory safety function.